keystorage/aws/src/main/java/tech/pegasys/signers/aws/AwsSecretsManager.java

```diff
@@ -103,40 +116,47 @@ private ListSecretsIterable listSecrets(
       final Collection<String> namePrefixes,
       final Collection<String> tagKeys,
       final Collection<String> tagValues,
       final BiFunction<String, String, R> mapper) {
     final Set<R> result = ConcurrentHashMap.newKeySet();
-    listSecrets(namePrefixes, tagKeys, tagValues)
-        .iterator()
-        .forEachRemaining(
-            listSecretsResponse -> {
-              listSecretsResponse
-                  .secretList()
-                  .parallelStream()
-                  .forEach(
-                      secretEntry -> {
-                        try {
-                          final Optional<String> secretValue = fetchSecret(secretEntry.name());
-                          if (secretValue.isEmpty()) {
+    final AtomicInteger errorCount = new AtomicInteger(0);
+    try {
+      listSecrets(namePrefixes, tagKeys, tagValues)
+          .iterator()
+          .forEachRemaining(
+              listSecretsResponse -> {
+                listSecretsResponse
+                    .secretList()
+                    .parallelStream()
+                    .forEach(
+                        secretEntry -> {
+                          try {
+                            final Optional<String> secretValue = fetchSecret(secretEntry.name());
+                            if (secretValue.isEmpty()) {
+                              LOG.warn(
+                                  "Failed to fetch secret name '{}', and was discarded",
+                                  secretEntry.name());
+                            } else {
+                              result.addAll(
+                                  mapSecretValue(mapper, secretEntry.name(), secretValue.get()));
+                            }
+                          } catch (final Exception e) {
                             LOG.warn(
-                                "Failed to fetch secret name '{}', and was discarded",
-                                secretEntry.name());
-                          } else {
-                            result.addAll(
-                                mapSecretValue(mapper, secretEntry.name(), secretValue.get()));
+                                "Failed to map secret '{}' to requested object type due to: {}.",
+                                secretEntry.name(),
+                                e.getMessage());
+                            errorCount.incrementAndGet();
                           }
-                        } catch (final Exception e) {
-                          LOG.warn(
-                              "Failed to map secret '{}' to requested object type due to: {}.",
-                              secretEntry.name(),
-                              e.getMessage());
-                        }
-                      });
-            });
-    return result;
+                        });
+              });
+    } catch (final Exception e) {
+      LOG.warn("Unexpected error during AWS list-secrets operation", e);
```

Comment:
What errors does this cover that the other try catch doesn't cover?

Reply:
this covers if the call of listSecret fail. Which is most likely to happen due to connectivity or credentials.